adafruit_dotstar.py

```diff
@@ -215,7 +218,20 @@ def brightness(self):
 
     @brightness.setter
     def brightness(self, brightness):
-        self._brightness = min(max(brightness, 0.0), 1.0)
+        brightness = min(max(brightness, 0.0), 1.0)
+        # scaling factor - new brightness / old.
+        # ex: going from 0.2 to 1 would be 1/0.2, so 5.0 * each color value
+        correction_factor = brightness / self.brightness
```

Comment:
While testing I discovered the 'brightness=0' edge case, which means that we'd lose every color on the strip.